Scrap the LivestockPrice.before we scrap any website or data we first need to import the API.we import the 'requests' where 'requests'is the webAPI.we import that requests.it will go to particular url and make a requests to access the data.next we import the beautifulsoup

It helps to parse the data from url using bs4(beautifulsoup4).it requests the urls.bs4 needs to parse the data from the url.

In [1]:
import requests

In [2]:
from bs4 import BeautifulSoup

we scarp the data from stock website(i.e)yahoo finance where there is a stock amazon.There are many stocks listed.we scrap here 'Amazon price' .we gonna to use this 'amazon' a ticker.we can call it as a ticker.so'amazon' is a ticker for the 'amazon' stock.that we store in variable 'stock'

In [3]:
stock='AMZN'
url='https://finance.yahoo.com/quote/'+str(stock)+'?p='+str(stock)
url

'https://finance.yahoo.com/quote/AMZN?p=AMZN'

here AMZN in url is the ticker value for the website .e.g: https://finance.yahoo.com/quote/GOOG/
                                    so wherever there is AMZN  is coming there we are having the stock and we are converting the stock into string

we can't pass AMZN value directly in url.its a good practice to store 'AMZN' (ticker) into a variable and then use it.so later if we want to scrap another stock price so we can directly change it for variable value.only one time.we dont need to change more times.

In [4]:
res=requests.get(url)
res

<Response [429]>

In [11]:
!pip install selenium

  Using cached h11-0.16.0-py3-none-any.whl.metadata (8.3 kB)
   ---------------------------------------- 0.0/9.4 MB ? eta -:--:--
   ----------------------------------- ---- 8.4/9.4 MB 47.2 MB/s eta 0:00:01
   ---------------------------------------- 9.4/9.4 MB 41.8 MB/s eta 0:00:00
Using cached h11-0.16.0-py3-none-any.whl (37 kB)

   ---- -----------------------------------  1/10 [websocket-client]
   ------------ ---------------------------  3/10 [h11]
   ---------------- -----------------------  4/10 [attrs]
   ------------------------ ---------------  6/10 [outcome]
   ---------------------------- -----------  7/10 [trio]
   ---------------------------- -----------  7/10 [trio]
   ---------------------------- -----------  7/10 [trio]
   ---------------------------- -----------  7/10 [trio]
   ---------------------------- -----------  7/10 [trio]
   ---------------------------- -----------  7/10 [trio]
   ---------------------------- -----------  7/10 [trio]
   ---------------------

In [12]:
import pandas as pd
from selenium import webdriver

Using selenium

In [ ]:
from selenium import webdriver
from selenium.webdriver.edge.service import Service
from selenium.webdriver.edge.options import Options

# Optional: Set up Edge options (headless mode, etc.)
edge_options = Options()
# edge_options.add_argument("--headless")  # Uncomment to run in headless mode

# Path to your msedgedriver.exe
edge_driver_path = "C:\\path\\to\\msedgedriver.exe"

# Initialize Edge WebDriver
service = Service(edge_driver_path)
driver = webdriver.Edge(service=service, options=edge_options)

# Open a URL
driver.get("https://www.example.com")

# Optional: Print the page title
print(driver.title)

# Don't forget to close the browser when done
driver.quit()


In [15]:
pip install webdriver-manager



   -------------------- ------------------- 1/2 [webdriver-manager]
   ---------------------------------------- 2/2 [webdriver-manager]

Note: you may need to restart the kernel to use updated packages.


In [27]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager

# Setup Chrome options
options = Options()
# options.add_argument('--headless')  # Optional: Run without UI

# Correct usage with service
service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service, options=options)

# Example usage
ticker = "AMZN"
driver.get(f"https://query1.finance.yahoo.com/v7/finance/quote?symbols={ticker}")

print(driver.title)
#driver.quit()


In [22]:
pip install yfinance


  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
     ---------------------------------------- 0.0/949.2 kB ? eta -:--:--
     ------------------------------------- 949.2/949.2 kB 42.9 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
   ---------------------------------------- 0.0/1.6 MB ? eta -:--:--
   ---------------------------------------- 1.6/1.6 MB 43.5 MB/s eta 0:00:00
  Created wheel for multitasking: filename=multitasking-0.0.12-py3-none-any.whl size=15703 sha256=65292b7ed4042b82b30f68fa3858fbbd0176d5ec8c3196f6e214f95032372019
  Stored in directory: c:\users\welcome\appdata\local\pip\cache\wheels\e9\25\85\25d2e1cfc0ece64b930b16972f

  DEPRECATION: Building 'multitasking' using the legacy setup.py bdist_wheel mechanism, which will be removed in a future version. pip 25.3 will enforce this behaviour change. A possible replacement is to use the standardized build interface by setting the `--use-pep517` option, (possibly combined with `--no-build-isolation`), or adding a `pyproject.toml` file to the source tree of 'multitasking'. Discussion can be found at https://github.com/pypa/pip/issues/6334


In [23]:
import yfinance as yf

ticker = yf.Ticker("AMZN")
data = ticker.info  # or use `ticker.history(period="1d")` for price history

print(data["currentPrice"])  # Or any other field like 'marketCap', 'fiftyTwoWeekHigh', etc.


213.75


In [25]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
from bs4 import BeautifulSoup
import time

# Setup Selenium
options = Options()
options.add_argument("--headless")  # optional
service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service, options=options)

# Navigate to Yahoo Finance page
ticker = "AMZN"
url = f"https://finance.yahoo.com/quote/{ticker}"
driver.get(url)

time.sleep(3)  # Wait for page to load

# Parse HTML using BeautifulSoup
soup = BeautifulSoup(driver.page_source, "html.parser")
price_tag = soup.find("fin-streamer", {"data-field": "regularMarketPrice"})

if price_tag:
    print(f"Current price of {ticker}: {price_tag.text}")
else:
    print("Price not found.")

#driver.quit()


Current price of AMZN: 
57.26



In [28]:
import time
import requests

ticker="AMZN"

#for ticker in tickers:
url = f"https://query1.finance.yahoo.com/v7/finance/quote?symbols={ticker}"
response = requests.get(url)
response
    
# Wait 2 seconds between requests
time.sleep(2)


Use caching—don’t re-fetch the same data repeatedly.Response 429 after we requests the url is blocking us to scrap.the response 200 usually is good to scrap.to avoid blocking we addded time delay

In [33]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
import time

# Setup driver
options = Options()
# options.add_argument('--headless')  # Uncomment if needed
service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service, options=options)

# Visit Yahoo Finance
ticker = "AMZN"
driver.get(f"https://finance.yahoo.com/quote/{ticker}")
time.sleep(3)

# Use correct method for locating elements (Selenium 4+)
price_elements = driver.find_elements(By.XPATH, '//div[@class="container yf-16vvaki"]//div[1]//span')

# Print found prices
for elem in price_elements:
    print("Price:", elem.text)

driver.quit()


Price: 213.75
Price: 215.17


In [34]:
#we have now one price if we want to capture live price whenever 
#it changes we want to capture live price.

from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
import time

def live_price():
    # Setup driver
    options = Options()
    # options.add_argument('--headless')  # Uncomment if needed
    service = Service(ChromeDriverManager().install())
    driver = webdriver.Chrome(service=service, options=options)

    # Visit Yahoo Finance
    ticker = "AMZN"
    driver.get(f"https://finance.yahoo.com/quote/{ticker}")
    time.sleep(3)

    # Use correct method for locating elements (Selenium 4+)
    price_elements = driver.find_elements(By.XPATH, '//div[@class="container yf-16vvaki"]//div[1]//span')

    # Print found prices
    for elem in price_elements:
         print("Price:", elem.text)

    return elem

    driver.quit()


stock_list=[]
while True:
    print('current price of '+str(stock)+'is=>'+str(live_price()))
    stock_list.append(str(live_price()))

#store in dataset
df=pd.DataFrame({'Stock':stock,'Price':stock_list})
df.head()

    

Price: 213.75
Price: 215.03
current price of AMZNis=><selenium.webdriver.remote.webelement.WebElement (session="030354df455696f05baecc60b6ed6c87", element="f.BB32227A32916E6FC05ABE2D08D3A3E1.d.76F2444D639C3186E1992333B4C442C7.e.97")>
Price: 213.75
Price: 214.95
Price: 213.75
Price: 215.00
current price of AMZNis=><selenium.webdriver.remote.webelement.WebElement (session="2700d0c751b499a4c22ecaeefe1b5724", element="f.A967FA4E114D19906C4FC0331F12E178.d.E6E49A7852B6E3325ED09D90E0DD454A.e.90")>
Price: 213.75
Price: 214.84
Price: 213.75
Price: 214.87
current price of AMZNis=><selenium.webdriver.remote.webelement.WebElement (session="634995615fc1404065272c2519953b54", element="f.36B3DF3D2BA86C513AF8EF56B5822097.d.1B51165AE0EC02260960934530D23DEC.e.94")>
Price: 213.75
Price: 214.87
Price: 213.75
Price: 214.87
current price of AMZNis=><selenium.webdriver.remote.webelement.WebElement (session="1c0ec055b03c575ebbbc8fc4035574bf", element="f.DC3B699C4804DAFF8AAA167D060BE83E.d.A1170BD82770CF1126157

WebDriverException: Message: unknown error: cannot determine loading status
from target frame detached
  (Session info: chrome=138.0.7204.184)
Stacktrace:
	GetHandleVerifier [0x0xadba83+63395]
	GetHandleVerifier [0x0xadbac4+63460]
	(No symbol) [0x0x921f70]
	(No symbol) [0x0x912b60]
	(No symbol) [0x0x910b44]
	(No symbol) [0x0x91160d]
	(No symbol) [0x0x91e3ac]
	(No symbol) [0x0x92f2b5]
	(No symbol) [0x0x934cb6]
	(No symbol) [0x0x911c4d]
	(No symbol) [0x0x92eb43]
	(No symbol) [0x0x9b0a9c]
	(No symbol) [0x0x98f1a6]
	(No symbol) [0x0x95e7b2]
	(No symbol) [0x0x95f654]
	GetHandleVerifier [0x0xd58883+2672035]
	GetHandleVerifier [0x0xd53cba+2652634]
	GetHandleVerifier [0x0xb02bca+223466]
	GetHandleVerifier [0x0xaf2cb8+158168]
	GetHandleVerifier [0x0xaf978d+185517]
	GetHandleVerifier [0x0xae3b78+96408]
	GetHandleVerifier [0x0xae3d02+96802]
	GetHandleVerifier [0x0xace90a+9770]
	BaseThreadInitThunk [0x0x76405d49+25]
	RtlInitializeExceptionChain [0x0x77dfd1ab+107]
	RtlGetAppContainerNamedObjectPath [0x0x77dfd131+561]


In [36]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
import time
import pandas as pd

# Define stock symbol
stock = "AMZN"
stock_list = []

# Setup Chrome driver once
options = Options()
# options.add_argument('--headless')  # Optional: Run without GUI
service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service, options=options)

# Open the stock page once
driver.get(f"https://finance.yahoo.com/quote/{stock}")
time.sleep(5)  # Let it fully load

# Collect 6 live prices
for i in range(6):
    try:
        # Get price from fin-streamer (more stable than div class)
        price_element = driver.find_element(By.XPATH, '//fin-streamer[@data-field="regularMarketPrice"]')
        price = price_element.text
        print(f"Price {i+1}: {price}")
        stock_list.append(price)
    except Exception as e:
        print("Error reading price:", e)
        stock_list.append("N/A")

    time.sleep(2)  # Delay between reads

# Quit browser
driver.quit()

# Save to DataFrame
df = pd.DataFrame({'Stock': [stock]*len(stock_list), 'Price': stock_list})
print(df.head())


Price 1: 
Price 2: 
Price 3: 
Price 4: 
Price 5: 
Price 6: 
  Stock Price
0  AMZN      
1  AMZN      
2  AMZN      
3  AMZN      
4  AMZN      


this means the XPath is matching an element, but its .text is empty. This is a common issue on JavaScript-heavy sites like Yahoo Finance where:

The price is loaded dynamically.

Even though the element is found, its value hasn't rendered yet.

✅ Fix: Use WebDriverWait to wait until the text content is not empty
We'll update your code to:

Wait until the fin-streamer[data-field="regularMarketPrice"] contains non-empty text.

Read the price.

Append it to the list.

In [37]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from webdriver_manager.chrome import ChromeDriverManager
import time
import pandas as pd

stock = "AMZN"
stock_list = []

# Setup WebDriver
options = Options()
# options.add_argument('--headless')  # Optional
service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service, options=options)

# Open Yahoo Finance stock page
driver.get(f"https://finance.yahoo.com/quote/{stock}")
time.sleep(5)  # Give time to load the JS framework

# Wait setup
wait = WebDriverWait(driver, 10)

# Capture 6 price updates
for i in range(6):
    try:
        # Wait until the price element has non-empty text
        price_element = wait.until(
            EC.presence_of_element_located((By.XPATH, '//fin-streamer[@data-field="regularMarketPrice"]'))
        )

        # Wait until the text is not empty
        price_text = ""
        retries = 0
        while price_text == "" and retries < 5:
            price_text = price_element.text.strip()
            if price_text != "":
                break
            time.sleep(1)
            retries += 1

        print(f"Price {i+1}: {price_text}")
        stock_list.append(price_text)

    except Exception as e:
        print("Error fetching price:", e)
        stock_list.append("N/A")

    time.sleep(2)  # Wait between samples

driver.quit()

# Save to DataFrame
df = pd.DataFrame({'Stock': [stock]*len(stock_list), 'Price': stock_list})
print(df)


Price 1: 
Price 2: 
Price 3: 
Price 4: 
Price 5: 
Price 6: 
  Stock Price
0  AMZN      
1  AMZN      
2  AMZN      
3  AMZN      
4  AMZN      
5  AMZN      


 if you're still getting no values, we’ll apply a bulletproof approach to fix it. Yahoo Finance uses Shadow DOM and delayed JavaScript rendering, so even the right element might be invisible until after page render.

Let’s use a tested and reliable XPath and explicitly wait for the text content of the fin-streamer[data-field="regularMarketPrice"], which always holds the live price.



In [38]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from webdriver_manager.chrome import ChromeDriverManager
import time
import pandas as pd

# Configuration
stock = "AMZN"
stock_list = []

# Setup Chrome
options = Options()
# options.add_argument('--headless')  # Optional
options.add_argument("--disable-blink-features=AutomationControlled")
options.add_argument("--start-maximized")
service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service, options=options)

# Open Yahoo Finance stock page
url = f"https://finance.yahoo.com/quote/{stock}"
driver.get(url)

# Wait for the regularMarketPrice element to appear
wait = WebDriverWait(driver, 15)

# Loop to collect 6 price updates
for i in range(6):
    try:
        # Wait until the live price is available AND has non-empty text
        price_element = wait.until(EC.presence_of_element_located(
            (By.XPATH, '//fin-streamer[@data-field="regularMarketPrice"]')
        ))

        # Retry if text is not loaded yet
        retries = 0
        price = price_element.text.strip()
        while not price and retries < 5:
            time.sleep(1)
            price = price_element.text.strip()
            retries += 1

        if price:
            print(f"Price {i+1}: {price}")
            stock_list.append(price)
        else:
            print(f"Price {i+1}: NOT FOUND")
            stock_list.append("N/A")

    except Exception as e:
        print(f"Error fetching price {i+1}:", e)
        stock_list.append("N/A")

    time.sleep(2)  # wait between price checks

# Clean up
driver.quit()

# Save to DataFrame
df = pd.DataFrame({'Stock': [stock]*6, 'Price': stock_list})
print(df)


Price 1: NOT FOUND
Price 2: NOT FOUND
Price 3: NOT FOUND
Price 4: NOT FOUND
Price 5: NOT FOUND
Price 6: NOT FOUND
  Stock Price
0  AMZN   N/A
1  AMZN   N/A
2  AMZN   N/A
3  AMZN   N/A
4  AMZN   N/A
5  AMZN   N/A


this confirms that Yahoo is actively delaying or hiding the fin-streamer content from headless or automation-like browsers.

Let’s now go step-by-step and use a reliable alternative method:

✅ Alternative: Use BeautifulSoup with Selenium to parse the rendered HTML
We’ll:

Use Selenium to render the full page (JavaScript included).

Use BeautifulSoup to parse the HTML.

Extract the current price from the fin-streamer tag reliably.

In [39]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.options import Options
from bs4 import BeautifulSoup
from webdriver_manager.chrome import ChromeDriverManager
import time
import pandas as pd

stock = "AMZN"
stock_list = []

# Set up Chrome options
options = Options()
# options.add_argument('--headless')  # Keep it visible for dynamic JS
options.add_argument("--disable-blink-features=AutomationControlled")
options.add_argument("--start-maximized")

# Set up Chrome driver
service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service, options=options)

# Open the stock page
url = f"https://finance.yahoo.com/quote/{stock}"
driver.get(url)
time.sleep(5)  # Wait for JS to render

# Collect 6 live prices
for i in range(6):
    try:
        # Refresh the page source
        soup = BeautifulSoup(driver.page_source, "html.parser")

        # Extract from fin-streamer
        price_tag = soup.find("fin-streamer", {"data-field": "regularMarketPrice"})
        price = price_tag.text.strip() if price_tag else "N/A"

        print(f"Price {i+1}: {price}")
        stock_list.append(price)
    except Exception as e:
        print(f"Error fetching price {i+1}:", e)
        stock_list.append("N/A")

    time.sleep(2)

driver.quit()

# Save to DataFrame
df = pd.DataFrame({'Stock': [stock]*6, 'Price': stock_list})
print(df)


Price 1: 118.32
Price 2: 118.32
Price 3: 118.32
Price 4: 118.32
Price 5: 118.32
Price 6: 118.32
  Stock   Price
0  AMZN  118.32
1  AMZN  118.32
2  AMZN  118.32
3  AMZN  118.32
4  AMZN  118.32
5  AMZN  118.32


Troubleshooting Tips:
Don't use headless mode if you’re getting blank data.

Make sure page fully loads before parsing.

Yahoo may A/B test layouts — fin-streamer is usually the most reliable.



In [40]:
#using beautiful soup
import requests

In [41]:
from bs4 import BeautifulSoup

In [42]:
stock1='AMZN'
url='https://finance.yahoo.com/quote/'+str(stock)+'?p='+str(stock1)
url

'https://finance.yahoo.com/quote/AMZN?p=AMZN'

In [43]:
res=requests.get(url)
res

<Response [429]>

In [48]:
import requests
from bs4 import BeautifulSoup

# Get the webpage
url = "https://finance.yahoo.com/quote/AMZN"
headers = {"User-Agent": "Mozilla/5.0"}  # Fake user agent to avoid blocking
res = requests.get(url, headers=headers)

# Parse the page
soup = BeautifulSoup(res.text, "html.parser")

# Try to locate the outer container
container = soup.find('div', {'class': "container yf-16vvaki"})

if container:
    # Look for the nested div > span inside it
    nested_div = container.find('div')
    if nested_div:
        price_span = nested_div.find('span')
        if price_span:
            print("✅ Price:", price_span.text)
        else:
            print("❌ Price span not found")
    else:
        print("❌ Nested div not found")
else:
    print("❌ Container div not found")


✅ Price: 213.75 


.find() directly on the result of .find_all(), which returns a list, not a single element.



In [50]:
import requests
from bs4 import BeautifulSoup

# Get the webpage
url = "https://finance.yahoo.com/quote/AMZN"
headers = {"User-Agent": "Mozilla/5.0"}  # Fake user agent to avoid blocking
res = requests.get(url, headers=headers)
def live_price():
    # Parse the page
    soup = BeautifulSoup(res.text, "html.parser")

    # Try to locate the outer container
    container = soup.find('div', {'class': "container yf-16vvaki"})

    if container:
         # Look for the nested div > span inside it
         nested_div = container.find('div')
         if nested_div:
             price_span = nested_div.find('span')
             if price_span:
                  print("✅ Price:", price_span.text)
             else:
                  print("❌ Price span not found")
         else:
              print("❌ Nested div not found")
    else:
          print("❌ Container div not found")
    return price_span.text


stock_list=[]
for i in range(6):
    print('current price of'+str(stock1)+'is=>'+str(live_price()))
    stock_list.append(str(live_price()))


# Save to DataFrame
df = pd.DataFrame({'Stock': [stock]*6, 'Price': stock_list})
print(df)


✅ Price: 213.75 
current price ofAMZNis=>213.75 
✅ Price: 213.75 
✅ Price: 213.75 
current price ofAMZNis=>213.75 
✅ Price: 213.75 
✅ Price: 213.75 
current price ofAMZNis=>213.75 
✅ Price: 213.75 
✅ Price: 213.75 
current price ofAMZNis=>213.75 
✅ Price: 213.75 
✅ Price: 213.75 
current price ofAMZNis=>213.75 
✅ Price: 213.75 
✅ Price: 213.75 
current price ofAMZNis=>213.75 
✅ Price: 213.75 
  Stock    Price
0  AMZN  213.75 
1  AMZN  213.75 
2  AMZN  213.75 
3  AMZN  213.75 
4  AMZN  213.75 
5  AMZN  213.75 


In [51]:
import requests
from bs4 import BeautifulSoup
import pandas as pd
import time

url = "https://finance.yahoo.com/quote/AMZN"
headers = {"User-Agent": "Mozilla/5.0"}

def live_price():
    res = requests.get(url, headers=headers)
    soup = BeautifulSoup(res.text, "html.parser")

    # Find price from reliable class or tag
    price_span = soup.find('fin-streamer', {'data-field': 'regularMarketPrice'})
    if price_span:
        return price_span.text.strip()
    else:
        print("❌ Price not found")
        return None

# Collect prices
stock = "AMZN"
stock_list = []

for i in range(6):
    price = live_price()
    print(f"Current price of {stock} is => {price}")
    stock_list.append(price)
    time.sleep(1)  # add delay to avoid getting blocked

# Save to DataFrame
df = pd.DataFrame({'Stock': [stock]*6, 'Price': stock_list})
print(df)


Current price of AMZN is => 6,341.75
Current price of AMZN is => 6,341.75
Current price of AMZN is => 6,341.75
Current price of AMZN is => 6,341.75
Current price of AMZN is => 6,341.75
Current price of AMZN is => 6,341.75
  Stock     Price
0  AMZN  6,341.75
1  AMZN  6,341.75
2  AMZN  6,341.75
3  AMZN  6,341.75
4  AMZN  6,341.75
5  AMZN  6,341.75


we can also give timer,we can also say whenever the price changes then only we print it.